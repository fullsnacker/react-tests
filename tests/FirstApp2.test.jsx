import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => { 

    const title = 'Hola, soy goku';

    test('debe hacer match con el snapshot', () => { 
        
        const { container } = render( <FirstApp title={ title }/>)

        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Hola, soy goku"', () => { 

        render( <FirstApp title={ title }/>)

        // screen.debug();

        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', () => { 

        render( <FirstApp title={ title }/>)

        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );
    });

})