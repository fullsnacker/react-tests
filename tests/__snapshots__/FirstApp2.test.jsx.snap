// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
